File: geometryFundamental/SubdivisionCurve/subdivisionCurve.cs
  private void RunScript(List<Point3d> points, int count, double quarter, ref object subdividePoints, ref object curve)
  {
    for (int j = 0; j < count; j++)
    {
      List<Point3d> pts = new List<Point3d>();

      for (int i = 0; i < points.Count - 1; i++)
      {
        Point3d quarterPoint1 = (1 - quarter) * points[i] + quarter * points[i + 1];
        pts.Add(quarterPoint1);

        Point3d quarterPoint2 = quarter * points[i] + (1 - quarter) * points[i + 1];
        pts.Add(quarterPoint2);
      }
      points = pts;
    }

    subdividePoints = points;
    curve = new PolylineCurve(points);

File: geometryFundamental/SubdivisionCurve/subdivisionCurve.py
import Rhino.Geometry as rg
import rhinoscriptsyntax as rs

quarter = 0.25
for j in range(count):
    pts = []
    for i in range(len(points)-1):
        x1, y1, z1 = ((points[i+1] - points[i]) * quarter) + points[i]
        pt_1 = rg.Point3d(x1, y1, z1)
        pts.append(pt_1)
        
        x2, y2, z2 = ((points[i+1] - points[i]) * (1 - quarter)) + points[i]
        pt_2 = rg.Point3d(x2, y2, z2)
        pts.append(pt_2)
    points = pts

subdiv_points = pts
subdiv_curve = rs.AddPolyline(points)
